### ISSUE TITLE
[2023] From Molecules to Materials: Pre-training Large Generalizable Models for Atomic Property Prediction

### ISSUE THREAD

**Issue by mjrs33** [mjrs33/paper#108]
## 概要
機械学習ポテンシャルの基盤モデルを作る研究。様々なドメインの複数のデータセットでマルチタスクの事前学習を行うことで、下流タスクの性能がロバストに向上する。これはID, OOD共に有効。データの不均衡を解消するロス関数の設定が精度向上に重要。単一の巨大なデータセットのみで学習した場合は、複数のデータセットを用いた場合より下流タスクの精度は低下する。

### 論文リンク
https://arxiv.org/abs/2310.16802

### 著者/所属機関
Nima Shoghi and Adeesh Kolluru and John R. Kitchin and Zachary W. Ulissi and C. Lawrence Zitnick and Brandon M. Wood

### 投稿日付
2023/10/25

## 背景
機械学習ポテンシャルのデータセット構築には時間がかかる。例えばOpen Catalyst 2020 (OC20)は400 million CPU hours必要であった。そのため、化学の全てのドメインに対してデータセットを作ることは現実的でない。

自然言語分野などで、基盤モデルの活用が進んでいるが、物性予測においてはあまり取り組みがなく、スクラッチで学習しSOTA性能を得ていることが多い。

有限で離散的なデータである自然言語や画像と異なり、原子の物性は連続的で数桁の範囲に及ぶという特有の問題がある。
そのためモデルは、外挿領域に対しロバストであり、また、変動しやすい出力を予測できる必要がある。

## 新規性
Joint Multi-domain Pre-training (JMP)を提案。これは教師ありの事前学習であり、様々な化学ドメインのデータセット群を同時に、マルチタスクで学習する。
様々な下流タスクで事前学習の効果を検証した。

機械学習ポテンシャルの事前学習では平衡状態の構造のデノイジング（forceがゼロに近い）を行うことがあるが、データセットのほとんどの構造は非平衡状態であり適用できない。本手法は非平衡状態の構造も用いることができる。

## 手法
全体像は以下。

![DE39FF5F-91F0-430D-9EC4-F20DEE7D1DCA](https://github.com/mjrs33/paper/assets/40561031/6748438c-3393-4144-87c7-300d7f05ed01)

4つのデータセットで事前学習を行い、様々な下流タスクで評価した。下流タスクは様々な化学ドメインのタスクで評価した。

### データセット
用いたデータセットは以下。

![2415DDD2-620C-4433-8037-9796BFBA8EB9](https://github.com/mjrs33/paper/assets/40561031/142c6cd2-4fab-49c2-b57a-cca7d2612fbd)

小分子（1-20原子）、大分子（20原子以上）、結晶構造、触媒（結晶の表面に分子がいる）の4つのドメインを設定。

事前学習のデータ数は120M以上であり、それぞれenergyとforceのラベルが付与されている。99%以上は非平衡状態の構造である。

下流タスクは異なるドメインのデータセットや、out-of-distribution (OOD)のラベルも含まれる。

### Joint multi-domain pre-training
JMPでは異なるデータセットのプロパティをマルチタスクで同時に学習させる。これにより様々なドメインの物性を１つのモデルで学習でき、良い表現学習ができる。

#### モデル
バックボーンとしてGemNet-OCを用いた。energyとforceはそれぞれノード特徴量、エッジ特徴量から得た。
ここで、forceはenergyの勾配で計算せず、直接算出している。これは計算コスト的に有利であり、大きなデータセットでは勾配で算出する方法と同程度の精度をより小さいepoch数で実現できるため採用したとのこと。
forceはE(3)-equivariantな予測でないといけないことに注意（モデルが制限される）。

#### 正規化
データセット間で似たようなスケールに収まるように目的変数を正規化する必要がある。
energyはデータセットごとにlinearly reference後に正規分布に変換したと書いてある。linearly referenceはおそらく、原子ごとに重みを持つ線形モデルでエネルギーをフィッティングしたもの。

forceはそれぞれの要素ごとのRMSで割ることで正規化した。

#### 不均衡なデータセットサイズへの対策
データ数が多いデータセットに過剰に適合することを防ぐため、バッチ作成時にtemperature samplingを導入した。
これはデータセットの比率を（1/T）乗した確率でサンプリングすることになる。今回はT=2を採用した。

#### 不均衡なシステムサイズへの対策
データセットごとに原子数の平均が異なるため、以下のようにロス関数を変更した。

![2B0CDB16-5379-4A33-A863-06694283246A](https://github.com/mjrs33/paper/assets/40561031/f87e10a1-e55c-4852-a622-48ee1c480dee)

赤+黒が通常のロス関数であり、原子ごとのforceを算出し、バッチ内全原子数で割っている。これは原子数が大きいデータセットが支配的になってしまう。
そのため、赤を削除し、青を追加した。これにより原子数が異なっていてもおおよそ同じ寄与度になる。この変更はモデルの性能にかなり影響したとのこと。

また、それぞれの重みλはenergyは1で固定し、forceはデータセットごとの平均原子数とした（つまりforceはそれぞれのデータセットごとに異なる重みで算出される）。

#### fine-tuning
学習済みモデルの予測ヘッドを、ランダムに初期化された新しいヘッドに変更し、モデル全体を下流タスクで学習させた。
下流タスクでforceを予測する場合はenergyの勾配から算出した。

## 結果
それぞれの下流タスクについて、スクラッチで学習した場合と比較して、事前学習した場合の改善率を示した結果が以下。

![9ECA35A8-48C2-4825-8529-22E106CFF954](https://github.com/mjrs33/paper/assets/40561031/dcd3fda9-1495-4bf3-8257-7b95a00f34eb)

まず、スクラッチで大きいモデルを学習した場合は、小さいモデルより性能が悪く、過学習が発生していることがわかる。
しかし、図中央をみると、事前学習により、大きなモデルの方が性能が高くなっており、小さなデータセットに巨大なモデルを適用しても過学習が起こらないことを示している。

スクラッチで学習した場合と比較して、事前学習した場合は平均で59%の精度改善が見られた。

QM9の予測結果は以下。

![A8142016-6CC2-4FD3-A9D3-2BDA456DF429](https://github.com/mjrs33/paper/assets/40561031/4e320005-0f3e-4ede-8065-55f5b1352c10)

ほぼ全てのタスクで本手法が最も良い性能であった。事前学習なしと比較すると全てのタスクで性能が向上している。

また、rMD17でも10個中5個でSOTA性能であり、残りもほぼ同等であった。50サンプルのfew-shotではさらに優位性が出るとのこと。

また、結晶材料に対しても有効性を確認。

![5B61BE8D-008B-4E83-AC72-4291E8B4B3BF](https://github.com/mjrs33/paper/assets/40561031/cae72200-d270-49b8-b138-bfd28db6da3e)

結晶構造はドメイン外のデータ、ラベルであり外挿性能も高いことが分かる。
この傾向は大分子に対しても同様であった。

### ablation study
ablation studyの結果は以下。

![47C5E59E-E203-4B39-AEE9-95D3C2BF6395](https://github.com/mjrs33/paper/assets/40561031/57b5d393-2824-4893-ba67-4336c50cbc41)

SWLはstructure wise loss averaging。
最も注目すべきはOC20だけで学習したときに精度が悪化する点である。ablation studyでは計算コストの都合上、サブセットを用いて事前学習をしている。しかしOC20は例外的に120Mのデータで学習した。
これが意味することは、事前学習のデータ数がかなり多いにも関わらず、単一のデータセットでは性能が向上しないということである。
そのため、多様なデータセットでマルチタスク学習をすることが重要である。
